packages/gen-ai-hub/src/orchestration/orchestration-client.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpDestination } from '@sap-cloud-sdk/connectivity';
+import { DefaultApi } from './api/default-api.js';
+import { CompletionPostRequest, CompletionPostResponse } from './api/schema/index.js';
+import { CustomRequestConfig } from '../core/http-client.js';
+
+export type GenAiHubCompletionParameters = Pick<
+  CompletionPostRequest,
+  'orchestration_config' | 'return_module_results'
+>;
+
+/**
+ * Get the orchestration client.
+ */
+export class GenAiHubClient {
+    destination: HttpDestination;
+    constructor(destination: HttpDestination) {
+        this.destination = destination;
```

Comment:
I would not expect the generated code can solve everything.
Instead, it should be used for saving some manual jobs, e.g., generating models.
For public API, I would anyway just create a wrapper, which should be stable (as the signature will not be updated automatically due to the spec).
The wrapper should be designed in the way, that is similar to the OpenAI client, as the orchestration service respects the OpenAI spec.

Reply:
on the other hand, requesting a spec improvement should also be considered as long term solution.

Reply:
I think the missing resource group is just an oversight in the spec and should (arguably must) be added there, regardless of what we do in the SDK. Do you want to raise this to them, or should I?

I recall the JS SDK had a shortcoming where it would not generate the resource group parameter. Is this still the case?

But yes, I think even if the SDK would generate correctly, it would probably? be part of the `DefaultAPI` (at least that is how it works in Java 😉 ). So regardless, I guess we take the same approach as in other places: send `default` by default (lol) and the user can override with arbitrary things, if needed..

Reply:
>I recall the JS SDK had a shortcoming where it would not generate the resource group parameter. Is this still the case?

The issue of generating header param has been fixed.

Reply:
@deekshas8 FYI I checked in with the colleagues, the header will be added to the spec 😉